### PR TITLE
remove redefinition of mean

### PR DESCRIPTION
**Pull request by kshitij12345**
Remove a redefinition of mean in pool.cc

### CODE REVIEW

**Comment by kshitij12345** [comment]
I was also wondering since mean is very much implemented, should we expose it to Python API and write some test cases for it against `numpy`.

File: chainerx_cc/chainerx/native/native_device/pool.cc [hunk 1]
```diff
@@ -220,7 +214,7 @@ class NativeAveragePoolForwardBackward : public chainerx::AveragePoolForwardBack
 
         switch (pad_mode_) {
             case AveragePoolPadMode::kZero:
-                Mean(col, kernel_axes, out);
+                out = Mean(col, kernel_axes);
```
Comment (hvy) [inline chainerx_cc/chainerx/native/native_device/pool.cc]: This `Mean` is a routine meaning that it has its own logic for deciding the output dtype (and connect the computational graph) which is not really what we want. Here we just want to compute the result.

**Comment by hvy** [comment]
Thanks the this PR! I wrote it in the comment but the `Mean` here is not a mere redefinition but introduced for a different purpose. As stated in the TODO, I proper solution would be to define `Device::Mean`.

**Comment by hvy** [comment]
On a different topic, exposing `Mean` and also `Var` with Python bindings would be great and indeed welcomed as PRs. :)

File: chainerx_cc/chainerx/native/native_device/pool.cc (hunk 1 above)
Comment (kshitij12345) [reply chainerx_cc/chainerx/native/native_device/pool.cc]: Oh, I see. I totally missed it. Thank you

**Comment by kshitij12345** [comment]
Oh see. Failed to notice that. Closing the PR for now. Will work on exposing to Python Api. And after that maybe try the device implementation for mean.

**Pull request closed without merging**